src/main/kotlin/no/nav/familie/ba/sak/integrasjoner/økonomi/UtbetalingsoppdragGeneratorService.kt

```diff
@@ -0,0 +1,173 @@
+package no.nav.familie.ba.sak.integrasjoner.økonomi
+
+import no.nav.familie.ba.sak.common.toYearMonth
+import no.nav.familie.ba.sak.kjerne.behandling.BehandlingHentOgPersisterService
+import no.nav.familie.ba.sak.kjerne.behandling.BehandlingService
+import no.nav.familie.ba.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ba.sak.kjerne.behandling.domene.BehandlingType
+import no.nav.familie.ba.sak.kjerne.behandling.domene.Behandlingsresultat
+import no.nav.familie.ba.sak.kjerne.beregning.BeregningService
+import no.nav.familie.ba.sak.kjerne.beregning.domene.AndelTilkjentYtelseRepository
+import no.nav.familie.ba.sak.kjerne.beregning.domene.TilkjentYtelseRepository
+import no.nav.familie.ba.sak.kjerne.vedtak.Vedtak
+import no.nav.familie.felles.utbetalingsgenerator.domain.BeregnetUtbetalingsoppdragLongId
+import no.nav.familie.felles.utbetalingsgenerator.domain.IdentOgType
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.YearMonth
+
+@Service
+class UtbetalingsoppdragGeneratorService(
+    private val behandlingHentOgPersisterService: BehandlingHentOgPersisterService,
+    private val behandlingService: BehandlingService,
+    private val tilkjentYtelseRepository: TilkjentYtelseRepository,
+    private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository,
+    private val utbetalingsoppdragGenerator: UtbetalingsoppdragGenerator,
+    private val beregningService: BeregningService,
+) {
+
+    fun genererUtbetalingsoppdrag(
+        vedtak: Vedtak,
+        saksbehandlerId: String,
+        erSimulering: Boolean = false,
+    ): BeregnetUtbetalingsoppdragLongId {
+        val forrigeBehandling =
+            behandlingHentOgPersisterService.hentForrigeBehandlingSomErIverksatt(behandling = vedtak.behandling)
+        val forrigeTilkjentYtelse =
+            forrigeBehandling?.let { tilkjentYtelseRepository.findByBehandlingAndHasUtbetalingsoppdrag(behandlingId = it.id) }
```

Comment:
Kanskje kan trekke ut disse 2 til
```suggestion
        val forrigeTilkjentYtelse = hentForrigeTilkjentYtelse(vedtak.behandling)
```


...
```kotlin
private fun hentForrigeTilkjentYtelse(behandling: Behandling): TilkjentYtelse? =
        behandlingHentOgPersisterService.hentForrigeBehandlingSomErIverksatt(behandling = behandling)
            ?.let { tilkjentYtelseRepository.findByBehandlingAndHasUtbetalingsoppdrag(behandlingId = it.id) }
```